File: ios/RnYandexnaviSignature.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(RnYandexnaviSignature, NSObject)

RCT_EXTERN_METHOD(sign:(NSString)url withClient:(NSString)client
                 withKey:(NSString)key withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/RnYandexnaviSignature.swift
@objc(RnYandexnaviSignature)
class RnYandexnaviSignature: NSObject {

    @objc(sign:withClient:withKey:withResolver:withRejecter:)
    func sign(url: String, client: String, key: String, resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
        reject("not support", "ios version is not supported yet", nil);
    }
}
